src/timeline/TimelineModel.cpp

```diff
@@ -1859,11 +1859,248 @@ TimelineModel::formatPowerLevelEvent(QString id)
+            // We do only compare to people with explicit PL. Usually others are not going to be
+            // affected either way and this is cheaper to iterate over.
+            for (auto const &[key, val] : event->content.users) {
+                if (val == event->content.kick) {
+                    number_of_affected++;
+                    if (number_of_affected <= 2) {
+                        affected.push_back(QString::fromStdString(key));
+                    }
+                }
+            }
+
+            if (number_of_affected != 0) {
+                auto true_affected_rest = number_of_affected - affected.size();
+                if (number_of_affected > 2) {
+                    return default_message + QString::fromStdString(" ") +
+                           tr("%1 and %2 and %n other(s) can now kick room members.")
+                             .arg(utils::replaceEmoji(displayName(affected.at(0))))
+                             .arg(utils::replaceEmoji(displayName(affected.at(1))))
+                             .arg(true_affected_rest);
+                } else if (number_of_affected == 2) {
+                    return default_message + QString::fromStdString(" ") +
+                           tr("%1 and %2 can now kick room members.")
+
+                             .arg(utils::replaceEmoji(displayName(affected.at(0))))
+                             .arg(utils::replaceEmoji(displayName(affected.at(1))));
+                } else if (number_of_affected == 1) {
+                    return default_message + QString::fromStdString(" ") +
+                           tr("%1 can now kick room members.")
+                             .arg(utils::replaceEmoji(displayName(affected.at(0))));
+                }
+            }
+        }
+
+        return default_message;
+    }
+
+    if (event->content.redact != prevEvent->content.redact) {
+        auto default_message = tr("%1 has changed the room's redact powerlevel from %2 to %3.")
+                                 .arg(sender_name)
+                                 .arg(prevEvent->content.redact)
+                                 .arg(event->content.redact);
+
+        // We only calculate affected users if we change to a level above the default users PL to
+        // not accidentially have a DoS vector
+        if (event->content.redact > default_powerlevel) {
+            QStringList affected{};
+            auto number_of_affected = 0;
+            // We do only compare to people with explicit PL. Usually others are not going to be
+            // affected either way and this is cheaper to iterate over.
+            for (auto const &[key, val] : event->content.users) {
+                if (val == event->content.redact) {
+                    number_of_affected++;
+                    if (number_of_affected <= 2) {
+                        affected.push_back(QString::fromStdString(key));
+                    }
+                }
+            }
```

Comment:
You could probably put that into a lambda, i.e.

```suggestion
    auto calc_affected = [&event](power_level_t level) -> std::pair<QStringList, int> {
            QStringList affected{};
            auto number_of_affected = 0;
            // We do only compare to people with explicit PL. Usually others are not going to be
            // affected either way and this is cheaper to iterate over.
            for (auto const &[key, val] : event->content.users) {
                if (val == event->content.redact) {
                    number_of_affected++;
                    if (number_of_affected <= 2) {
                        affected.push_back(QString::fromStdString(key));
                    }
                }
                return {affected, number_of_affected};
            };
```

And then call it like:

```cpp
auto [affected, number_of_affected] = calc_affected(event->content.users);
```

Reply:
\*_\* yes :3 makes sense :D